src/main/java/ubc/pavlab/rdp/util/TextUtils.java

```diff
@@ -0,0 +1,18 @@
+package ubc.pavlab.rdp.util;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class TextUtils {
+
+    public static List<String> tokenize( String query ) {
+        return Arrays.stream( query.replaceAll( "[^\\w-:]+", " " ).split( "\\s+" ) )
+                .map( String::toUpperCase )
+                .collect( Collectors.toList() );
+    }
+
+    public static String normalize( String query ) {
+        return String.join( " ", tokenize( query ) );
+    }
+}
```

Comment:
I'm not sure this utility class is worth having around since it's only used once. If we consider reusing some of the logic for parsing queries for autocompletion of GO terms too, then maybe yeah. See #174